### PR TITLE
feat: Grant forms' own url

### PR DESCRIPTION
**Pull request by ncomerci**
image

### CODE REVIEW

File: src/pages/submit/grant.tsx [hunk 1]
```diff
@@ -79,9 +80,30 @@ function parseStringsAsNumbers(grantRequest: GrantRequest) {
...
+  }
+}
+
 export default function SubmitGrant() {
   const t = useFormatMessage()
   const [account, accountState] = useAuthContext()
+  useEffect(() => {
+    const params = new URLSearchParams(location.search)
+    let category: NewGrantCategory | null = null
+    try {
```
Comment (andyesp) [inline src/pages/submit/grant.tsx:97]: Why don't...?
```
    try {
      const category = toNewGrantCategory(params.get('category'))
      patchGrantRequest({ category })
    } catch (error) {
      console.error(error)
    }
```

File: src/pages/submit/grant.tsx [hunk 2]
```diff
@@ -79,9 +80,30 @@ function parseStringsAsNumbers(grantRequest: GrantRequest) {
...
+      category = toNewGrantCategory(params.get('category'))
+    } catch (error) {
+      console.error(error)
+    } finally {
+      patchGrantRequest({ category })
+    }
+    // eslint-disable-next-line react-hooks/exhaustive-deps
```
Comment (andyesp) [inline src/pages/submit/grant.tsx:104]: I'd remove this and add the dependencies. Is params triggering a re-render?

File: src/pages/submit/grant.tsx (hunk 1 above)
Comment (ncomerci) [reply src/pages/submit/grant.tsx:97]: If `toNewGrantCategory` throws an error, the patch would not be executed and it is necessary that it is done regardless of whether it fails or not, that is why it is in finally

File: src/pages/submit/grant.tsx (hunk 2 above)
Comment (ncomerci) [reply src/pages/submit/grant.tsx:104]: image

It suggests adding the patch function to the dependencies, which is not necessary. That's why I omitted it

Comment (1emu) [reply src/pages/submit/grant.tsx:104]: Don't you need to add `location.search` to the deps?

**Review by andyesp — APPROVED** [review]

**Pull request merged**